app/services/ReturnsService.scala

```diff
@@ -33,81 +34,79 @@ class ReturnsService @Inject()(vatObligationsConnector: VatObligationsConnector,
+    vatObligationsConnector.getVatReturnObligations(user.vrn, Some(from), Some(currentDate), Status.Fulfilled).map {
+      case Right(obligations) => Right(obligations)
+      case Left(_) => Left(ObligationError)
+    }
+  }
 
   def getLastObligation(obligations: Seq[VatReturnObligation]): VatReturnObligation = obligations.sortWith(_.due isAfter _.due).head
 
   def getObligationWithMatchingPeriodKey(user: User, year: Int, periodKey: String)
-  (implicit hc: HeaderCarrier, ec: ExecutionContext): Future[Option[VatReturnObligation]] =
-  getReturnObligationsForYear(user, year, Status.Fulfilled).map {
-  case Right(VatReturnObligations(obligations)) => obligations.find(_.periodKey == periodKey)
-  case Left(_) => None
-}
+                                        (implicit hc: HeaderCarrier, ec: ExecutionContext): Future[Option[VatReturnObligation]] =
+    getReturnObligationsForYear(user, year, Status.Fulfilled).map {
+      case Right(VatReturnObligations(obligations)) => obligations.find(_.periodKey == periodKey)
+      case Left(_) => None
+    }
 
   private[services] def filterObligationsByDueDate(returnObligations: VatReturnObligations, searchYear: Int): VatReturnObligations =
-  VatReturnObligations(returnObligations.obligations.filter(_.end.getYear == searchYear))
+    VatReturnObligations(returnObligations.obligations.filter(_.end.getYear == searchYear))
 
   def getPayment(user: User, requiredPeriod: String, year: Option[Int] = None)
-  (implicit hc: HeaderCarrier, ec: ExecutionContext): Future[Option[Payment]] =
-  financialDataConnector.getPayments(user.vrn, year).map {
-  case Right(payments) => filterPaymentsByPeriodKey(payments, requiredPeriod)
-  case Left(_) => None
-}
+                (implicit hc: HeaderCarrier, ec: ExecutionContext): Future[Option[Payment]] = {
+    financialDataConnector.getPayments(user.vrn, year).map {
+      case Right(payments) => filterPaymentsByPeriodKey(payments, requiredPeriod)
+      case Left(_) => None
+    }
+  }
 
   private[services] def filterPaymentsByPeriodKey(payments: Payments, requiredPeriod: String): Option[Payment] =
-  payments.financialTransactions.find(_.periodKey == requiredPeriod)
+    payments.financialTransactions.find(_.periodKey == requiredPeriod)
 
   def constructReturnDetailsModel(vatReturn: VatReturn, payment: Option[Payment]): VatReturnDetails = {
-  val moneyOwed = payment.fold(true)(_.outstandingAmount != 0)
-  val isRepayment = vatReturn.vatReclaimedCurrentPeriod > vatReturn.totalVatDue
-  VatReturnDetails(vatReturn, moneyOwed, isRepayment, payment)
-}
-
-  def getDirectDebitStatus(vrn: String)(implicit hc: HeaderCarrier, ec: ExecutionContext): Future[ServiceResponse[Boolean]] = {
-  financialDataConnector.getDirectDebitStatus(vrn) map {
-  case Right(directDebitStatus) => Right(directDebitStatus)
-  case Left(_) => Left(DirectDebitStatusError)
-}
-}
+    val moneyOwed = payment.fold(true)(_.outstandingAmount != 0)
+    val oweHmrc: Option[Boolean] = payment map {
```

Comment:
Do we need this sort of logic around checking charge types? Now that we know the outstanding amount will only ever be negative when it's a repayment, we can use that figure to work this out and not have to care about charge type names.

Reply:
Thanks that's true.